lib/magma/actions/update_attribute.rb

```diff
@@ -0,0 +1,48 @@
+require 'magma/actions/action_error'
+
+class Magma
+  class UpdateAttributeAction
+    def initialize(project_name, action_params = {})
+      @project_name = project_name
+      @action_params = action_params
+      @errors = []
+    end
+
+    def perform
+      @action_params.slice(*Magma::Attribute::EDITABLE_OPTIONS).each do |option, value|
+        attribute.update_option(option, value)
+      rescue => e
```

Comment:
This broad rescue will probably get us into trouble; here we are catching everything and sending it back as a 422, "Bad Request". This may be correct, but the exception may instead be due to an error on our part, which should return as a 500. If we intend to use this rescue to catch 422 errors, we should rescue precisely only those errors that could be the result of user error and not just any exception. Although in general, I think my preference is usually to check for user errors by hand rather than relying on an exception being generated.

You are also packaging the message from the exception and returning it to the user. The raw error is useful for debugging and could go into the logs, but the user is not debugging the code, and does not know or need to know what the ruby class of the error was, etc. (in addition to whatever else the error might improperly reveal, who knows). For this reason I'd rather give the user a formatted response based on the error type, tailored to help them turn their "bad request" into a "good request", but not the raised exception itself.

Reply:
I think this should be addressed in a separate PR. I think the best way to accomplish this is with model-level validations on the attribute. We could